src/client/utils.ts

```diff
@@ -28,7 +28,14 @@ export function processHeaders(request: PublishRequest) {
   headers.set("Upstash-Method", request.method ?? "POST");
 
   if (request.delay !== undefined) {
-    headers.set("Upstash-Delay", `${request.delay.toFixed(0)}s`);
+    // Handle both string (Duration type) and number inputs for delay
+    if (typeof request.delay === "string") {
+      // If delay is a string (e.g., "20s", "1h"), use it directly
+      headers.set("Upstash-Delay", request.delay);
+    } else {
+      // If delay is a number, convert it to seconds and append 's'
+      headers.set("Upstash-Delay", `${request.delay.toFixed(0)}s`);
+    }
```

Comment:
Could it be beneficial to use a "standard" duration string parser, like [timestring](https://github.com/mike182uk/timestring/blob/main/index.js) or [parse-duration](https://github.com/jkroso/parse-duration/blob/master/index.js)?

Just copy the code over.

Reply:
Sounds good, but "ts" check is enough for now. QStash will thrown an exception if you send a malformed unit because its already being parsed on the server. They use a similar library.

Reply:
Ah, got it.